qiime/sdk/visualizer.py

```diff
@@ -220,8 +220,10 @@ def callable_wrapper(*args, **kwargs):
 
             # TODO use user-configured temp dir
             with tempfile.TemporaryDirectory('qiime2-temp-') as temp_dir:
-                # TODO make sure _callable doesn't return anything
-                self._callable(output_dir=temp_dir, **view_args)
+                ret_val = self._callable(output_dir=temp_dir, **view_args)
+                if ret_val is not None:
+                    raise ValueError("Visualizer function should not return "
```

Comment:
Can you make this a `TypeError`?

Can you also update the text to reference the problematic visualizer? Something using the repr should be pretty informative, e.g.

```python
raise TypeError("Visualizer %r should not return anything. Received %r as a return value." % (self, ret_val))
```